hazelcast-integration/spring-hibernate-2ndlevel-cache/src/main/java/com/hazelcast/hibernate/springhibernate2lc/persistence/BookRepository.java

```diff
@@ -1,9 +1,15 @@
 package com.hazelcast.hibernate.springhibernate2lc.persistence;
 
+import jakarta.persistence.QueryHint;
+import org.springframework.data.jpa.repository.QueryHints;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 
+import static org.hibernate.jpa.HibernateHints.HINT_CACHEABLE;
+
 @Repository
 public interface BookRepository extends CrudRepository<Book, Long> {
-
+    @Override
```

Comment:
This @QueryHint annotation is using Query Cache of Hibernate. It has nothing to do with second level cache of Hibernate that uses HZ as provider. I think it will be confusing for users.

Reply:
From: https://docs.jboss.org/hibernate/orm/6.0/userguide/html_single/Hibernate_User_Guide.html#caching-config-properties

> `hibernate.cache.use_query_cache`
> Enable or disable second level caching of query results. The default is false.

IMHO it's related to second level caching. Added comment https://github.com/hazelcast/hazelcast-code-samples/pull/576/commits/9f5e2b647c675fe464020c9de0af967e38bda1bc